app.js

```diff
@@ -0,0 +1,299 @@
+    [kubeCertificate],
+    async (certificateFilePath) => {
+      const [certificatePath, certificateFileName] = splitDirectory(certificateFilePath);
+      const certificateVolumeDefinition = docker.createVolumeDefinition(certificatePath);
+      const volumeDefinitions = [
+        certificateVolumeDefinition,
+      ];
+
+      const authenticationParameters = generateAuthenticationParameters({
+        certificateFilePath: `$${certificateVolumeDefinition.mountPoint.name}/${certificateFileName}`,
+        kubeToken: pluginParameters.kubeToken,
+        kubeApiServer: pluginParameters.kubeApiServer,
+        kubeUser: pluginParameters.kubeUser,
+      });
+
+      const dockerEnvironmentalVariables = volumeDefinitions.reduce(
+        (acc, cur) => ({
+          ...acc,
+          [cur.mountPoint.name]: cur.mountPoint.value,
+        }),
+        {},
+      );
+
+      const shellEnvironmentalVariables = volumeDefinitions.reduce(
+        (acc, cur) => ({
+          ...acc,
+          [cur.path.name]: cur.path.value,
+        }),
+        dockerEnvironmentalVariables,
+      );
+
+      const helmCommand = `\
+${sanitizeCommand(cliCommand)} \
+${authenticationParameters.join(" ")}`;
+
+      const command = docker.buildDockerCommand({
+        command: helmCommand,
+        image: HELM_IMAGE_NAME,
+        environmentVariables: dockerEnvironmentalVariables,
+        volumeDefinitionsArray: volumeDefinitions,
+      });
+
+      result = await exec(command, {
+        env: shellEnvironmentalVariables,
+      });
+    },
+  );
+
+  const { stdout, stderr } = result;
+
+  if (!stdout && stderr) {
+    throw new Error(stderr);
+  }
+
+  return stdout;
+}
+
+function generateAuthenticationParameters(pluginParams) {
+  const params = [];
+
+  params.push("--kube-ca-file", pluginParams.certificateFilePath);
+  params.push("--kube-token", pluginParams.kubeToken);
+  params.push("--kube-apiserver", pluginParams.kubeApiServer);
+  params.push("--kube-as-user", pluginParams.kubeUser);
+
+  return params;
+}
+
+function generateReleaseNameParameters({
+  releaseName,
+  generateName,
+}) {
+  if (releaseName && generateName) {
+    throw new Error("Please provide either Release Name or Generate Name parameter.");
+  }
+
+  if (!releaseName && !generateName) {
+    throw new Error("Please provide Release Name or Generate Name parameter.");
+  }
+
+  const params = [];
+
+  if (releaseName) {
+    params.push(releaseName);
+  }
+
+  if (generateName) {
+    params.push("--generate-name");
+  }
+
+  return params;
+}
+
+function generateInstallationParameters(pluginParams) {
+  const {
+    namespace,
+    valuesOverrides,
+  } = pluginParams;
+
+  const params = [];
+
+  if (namespace) {
+    params.push("--namespace", namespace);
+  }
+
+  if (valuesOverrides) {
+    params.push(valuesOverrides.map((override) => ["--set", override]).flat());
+  }
+
+  return params;
+}
+
+function splitDirectory(directory) {
+  const pathElements = directory.split("/");
```

Comment:
It's better to use `path` module for that:
```js
const fileName = pathModule.basename(directory);
const directoryPath = pathModule.dirname(directory);
```
or even
```js
return [
  pathModule.dirname(directory),
  pathModule.basename(directory),
];
```

Reply:
Fixed